File: Arrays/Maximum Subarray/kedane.js
function maxSubarraySum(arr) {
  let maximum = arr[0];
  let maxarray = arr[0];

  // Iterujemy po reszcie tablicy, rozpoczynając od drugiego elementu (indeks 1)
  for (let i = 1; i < arr.length; i++) {
    // Aktualizujemy maksymalną sumę kończącą się w bieżącym elemencie
    maxarray = Math.max(arr[i], maxarray + arr[i]);

    // Aktualizujemy maksymalną sumę do tej pory
    maximum = Math.max(maximum, maxarray);
  }

  return maximum;
}

// Przykład użycia
const arr = [-2, 1, -3, 4, -1, 2, 1, -5, 4];
const maxSum = maxSubarraySum(arr);
console.log("Maksymalna suma podciągu to:", maxSum);

File: Arrays/Maximum Subarray/kedane.py
class Solution:
    def maxSubArray(self, nums: list[int]) -> int:
        maximum = maxarray = nums[0]
        for i in range(1, len(nums)):
            # print(maxarray)
            maxarray = max(nums[i], maxarray + nums[i])
            maximum = max(maxarray, maximum)
        return maximum


array = [1, 2, 3, 6, -4, -5, 5]
solution_obj = Solution()
result = solution_obj.maxSubArray(array)
print("Maksymalna suma podciągu to:", result)
